paddle/fluid/pir/transforms/onednn/cpu_bfloat16_placement_pass.cc

```diff
@@ -0,0 +1,594 @@
+ public:
+  using pir::OpRewritePattern<OpType>::OpRewritePattern;
+
+  // find orphaned bfloat16 operator that is between two float32 operators
+  // revert mkldnn_data_type attr to float32
+  bool MatchAndRewrite(
+      OpType op,
+      pir::PatternRewriter &rewriter) const override {  // NOLINT
+
+    bool prev_fp32 = false;
+    bool next_fp32 = false;
+    const std::set<std::string> constant_op({"pd_op.data",
+                                             "builtin.parameter",
+                                             "pd_op.feed",
+                                             "pd_op.fetch",
+                                             "pd_op.assign"});
+
+    if (op->num_operands()) {
+      for (uint32_t i = 0; i < op->num_operands(); i++) {
+        if (!op->operand_source(i) || !op->operand_source(i).type()) {
+          continue;
+        }
+        auto *prev_op = pir::GetDefiningOpForInput(op, i);
+        // Some ops do not need to be processed
+        std::string prev_name = prev_op->name();
+        if (constant_op.count(prev_name)) {
+          continue;
+        }
+
+        // if (prev_op->HasTrait<pir::SideEffectTrait>() ||
+        // prev_op->isa<pir::ConstantTensorOp>() ||
+        // prev_op->isa<pir::ParameterOp>() ||
+        // prev_op->isa<paddle::dialect::FeedOp>() ||
+        // prev_op->isa<paddle::dialect::DataOp>()) {
+        //       continue;
+        //     }
+
+        auto op_attr = prev_op->attributes();
+        if (op_attr.find("mkldnn_data_type") == op_attr.end()) {
+          // data_type_is_missing
+          prev_fp32 = true;
+          break;
+        }
+        if (op_attr.find("mkldnn_data_type")->second == "float32") {
+          prev_fp32 = true;
+          break;
+        }
+      }
+    } else {
+      // The first op in graph
+      return false;
+    }
+
+    for (uint32_t i = 0; i < op->num_results(); i++) {
+      if (!op->result(i) || !op->result(i).type()) {
+        continue;
+      }
+      auto next_op_list = pir::GetUseOpsForOutput(op, i);
+      for (auto const &[next_op, op_index] : next_op_list) {
+        // Some ops do not need to be processed
+        std::string next_op_name = next_op->name();
+        if (constant_op.count(next_op_name)) {
+          continue;
+        }
+        auto op_next_attr = next_op->attributes();
+        if (op_next_attr.find("mkldnn_data_type") == op_next_attr.end()) {
+          // data_type_is_missing
+          VLOG(0) << "data_type_is_missing:" << next_op->name();
+          next_fp32 = true;
+          break;
+        }
+        if (op_next_attr.find("mkldnn_data_type")->second == "float32") {
+          VLOG(0) << "mkldnn_data_type is fp32:" << next_op->name();
+          next_fp32 = true;
+          break;
+        }
+      }
+    }
+
+    if (prev_fp32 && next_fp32) {
+      VLOG(0) << "RemoveOrphanedOperators:" << op->name();
+    } else {
+      VLOG(0) << "Not RemoveOrphanedOperators:" << op->name();
+      return false;
+    }
```

Comment:
Regardless of VLOG, How about using:
```
    if (!prev_fp32 || !next_fp32) return false;
```
for later version? Could be shorter and clearer.